File: tournaments/fractionSum/fractionSum.py
def fractionSum(a, b):
    c = a[1] * b[1]
    d = a[0] * b[1] + b[0] * a[1]
    gcd = getGcd(c, d)
    
    return [d/gcd, c/gcd]

def getGcd(a, b):
    while b:
        a, b = b, a % b
    return a

File: tournaments/fractionSum/fractionSum.js
function fractionSum(a, b) {
    const getGcd = function(a, b) {
        if (!a) return b;

        return getGcd(b % a, a);
    }

    const c = a[1] * b[1];
    const d = a[0] * b[1] + b[0] * a[1];
    const gcd = getGcd(c, d);

    return [d/gcd, c/gcd];
}
